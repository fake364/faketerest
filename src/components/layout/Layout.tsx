import React from "react";
import Header from "../header/Header";
import NavigationContainer from "../header/navigation/base-container/NavigationContainer";
import BaseHead from "../head/BaseHead";

type Props = {};

const Layout: React.FC<Props> = ({children}) => {
    return <>
        <BaseHead></BaseHead>
        <Header><NavigationContainer></NavigationContainer></Header>
        <main>{children}</main>
    </>
};

export default Layout;