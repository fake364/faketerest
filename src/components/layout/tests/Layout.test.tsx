import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../Layout';
import * as redux from 'react-redux';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn()
}));

describe('Layout', () => {
  it('should display Auth navigation if user is logged in', () => {
    const mockState = { metadata: { isLoggedIn: true } };
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((callback) => callback(mockState));
    const wrapper = shallow(<Layout>Text</Layout>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should display Common navigation if user is not logged in', () => {
    const mockState = { metadata: { isLoggedIn: false } };
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((callback) => callback(mockState));
    const wrapper = shallow(<Layout>Text</Layout>);
    expect(wrapper).toMatchSnapshot();
  });
});
