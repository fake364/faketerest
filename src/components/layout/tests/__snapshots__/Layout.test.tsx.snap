// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should display Auth navigation if user is logged in 1`] = `
<div>
  <BaseHead />
  <Header>
    <AuthNavigation />
  </Header>
  <main
    className="h-full"
  >
    Text
  </main>
</div>
`;

exports[`Layout should display Common navigation if user is not logged in 1`] = `
<div>
  <BaseHead />
  <Header>
    <NavigationContainer>
      <CommonNavigation />
    </NavigationContainer>
  </Header>
  <main
    className="h-full"
  >
    Text
  </main>
</div>
`;
